src/store/index.ts

```diff
@@ -0,0 +1,103 @@
+import { createStore, combineReducers, applyMiddleware } from 'redux';
+import { composeWithDevTools } from 'redux-devtools-extension';
+import { Dispatch } from 'react';
+import thunk from 'redux-thunk';
+import { getAllProducts } from '../helpers/api';
+import productsReducer, { setProducts } from './products';
+import paginationReducer from './pagination';
+import sortReducer from './sort';
+import loadingReducer from './loading';
+import queryReducer from './query';
+import favouritesReducer from './favourites';
+import cartReducer from './cart';
+import productReducer from './product';
+
+export const getPage = (state: RootState) => state.pagination.page;
+export const getPerPage = (state: RootState) => state.pagination.perPage;
+export const getSortField = (state: RootState) => state.sort;
+export const getQuery = (state: RootState) => state.query;
+export const getQuantity = (
+  state: RootState,
+  type: string,
+) => state.products
+  .filter((product: Products) => product.name.toLowerCase().includes(state.query.toLowerCase()))
+  .filter((product: Products) => product.type === type).length;
+
+export const getVisible = (state: RootState) => state.loading.isVisible;
+export const getLoading = (state: RootState) => state.loading.isLoading;
+export const getProducts = (state: RootState) => state.products;
+export const getFavourites = (state: RootState) => state.favourites;
+export const getProduct = (state: RootState) => state.product;
+
+
+const rootReducer = combineReducers({
+  products: productsReducer,
+  pagination: paginationReducer,
+  sort: sortReducer,
+  loading: loadingReducer,
+  query: queryReducer,
+  favourites: favouritesReducer,
+  cart: cartReducer,
+  product: productReducer,
+});
+
+
+export type RootState = ReturnType<typeof rootReducer>;
+
+
+export const getVisibleProducts = (state: RootState, type: string) => {
+  let compare: (a: Products, b: Products) => number = () => 0;
```

Comment:
do you return this or where do you use it?

Reply:
Linter solved the problem like this